modules/ROOT/pages/client-credentials.adoc

```diff
@@ -0,0 +1,297 @@
+
+[source, java, linenums]
+----
+public interface ClientCrendentialsState {
+
+  /**
+   * @return The obtained access token
+   */
+  String getAccessToken();
+
+  /**
+   * @return The access token's expiration. The actual format of it depends on the OAuth provider
+   */
+  Optional<String> getExpiresIn();
+}
+----
+
+Through this object, the provider gains access to the accessToken and other standard information that was obtained during the OAuth dance. The original Salesforce example shows how the connect() method makes use of this POJO to create the client.
+
+== Configure a Client Credentials Connector
+
+An important aspect of OAuth-enabled modules is how to use them. As much as the SDK does to hide the complexities of the OAuth protocol, things like the OAuth dance are unavoidable. The SDK’s approach to this problem is to standardize the experience of all OAuth modules to keep simplifying the user experience.
+This section discusses the steps a user must take to use the module.
+
+=== Synthetic Parameters
+In addition to everything explicitly defined in the ConnectionProvider, the SDK automatically adds some extra parameters and injects the proper behavior for them.
+
+[cols="1,1,1,2,2", options="header"]
+|===
+|Parameter Name
+|Required
+|Expressions
+|Default Value
+|Description
+
+|`clientId`
+|YES
+|SUPPORTED
+|N/A
+|The OAuth `clientId` as registered with the service provider.
+
+|`clientSecret`
+|YES
+|NOT_SUPPORTED
+|N/A
+|The OAuth `clientSecret` as registered with the service provider.
+
+|`tokenUrl`
+|NO
+|SUPPORTED
+|Value provided in the @ClientCredentials annotation.
+|The service provider's token endpoint URL.
+
+|`scopes`
+|NO
+|SUPPORTED
+|Value provided in the @ClientCredentials annotation.
+|The OAuth scopes to be requested during the dance. If not provided, it defaults to those in the annotation.
+
+|`objectStore`
+|NO
+|NOT_SUPPORTED
+|N/A
+|A reference to the ObjectStore used for storing each resource owner ID’s data. If not specified, the runtime automatically provisions the default one.
+|===
+
+=== About the Use of Expressions
+
+The previous table identifies many of the synthetic parameters that accept expressions. Using expressions there has the same effect as using expressions in a regular parameter: it turns the configuration into a dynamic one.
+
+=== OAuth Connection DSL
+
+The following example shows  generated DSL:
+
+[source, java, linenums]
+----
+<sfdc:config name="salesforce">
+    <sfdc:client-credentials-connection display="PAGE" immediate="FALSE" prompt="CONSENT">
+        <sfdc:oauth-client-credentials clientId="${sfdc.consumerkey}" clientSecret="${sfdc.consumersecret}" tokenUrl="http://..." />
+        <sfdc:oauth-store-config objectStore="oauthObjectStore" />
+</sfdc:config>
+----
+
+Regular and OAuth parameters are all shown at the connection provider level, just like in any other provider.
```

Comment:
This sentence compares parameters with "any other provider"; reword to clarify.

Reply:
```suggestion
Regular and OAuth parameters are all shown at the connection provider level, as they are in any other provider.
```